diff --git a/server/index.cjs b/server/index.cjs
index 7d82e39b1923dfbe266a7c689259af017e225cb4..dd46ac5633f639c7dadcd81234d1dd921ced7518 100644
--- a/server/index.cjs
+++ b/server/index.cjs
@@ -1223,7 +1223,8 @@ class UTApi {
                     id: S__namespace.String,
                     key: S__namespace.String,
                     name: S__namespace.String,
-                    status: S__namespace.Literal("Deletion Pending", "Failed", "Uploaded", "Uploading")
+                    status: S__namespace.Literal("Deletion Pending", "Failed", "Uploaded", "Uploading"),
+                    customId: S__namespace.NullOr(S__namespace.String)
                 }))
             });
             return await this.executeAsync(this.requestUploadThing("/api/listFiles", {
diff --git a/server/index.d.ts b/server/index.d.ts
index 51143242913a807825a06b0cf5678effe24e9177..ac3631ade2f5cb6f5265ff8ca032964e4a211223 100644
--- a/server/index.d.ts
+++ b/server/index.d.ts
@@ -194,6 +194,7 @@ declare class UTApi {
             readonly key: string;
             readonly status: "Deletion Pending" | "Failed" | "Uploaded" | "Uploading";
             readonly id: string;
+            readonly customId: string | null;
         }[];
     }>;
     renameFiles: (updates: RenameFileUpdate | RenameFileUpdate[]) => Promise<{
diff --git a/server/index.js b/server/index.js
index 22605f1fce9fe957874eaf7f9eeaa33dac1dd79d..49eb144387e98926f38fef3deba6d85ad74d5f80 100644
--- a/server/index.js
+++ b/server/index.js
@@ -1199,7 +1199,8 @@ class UTApi {
                     id: S.String,
                     key: S.String,
                     name: S.String,
-                    status: S.Literal("Deletion Pending", "Failed", "Uploaded", "Uploading")
+                    status: S.Literal("Deletion Pending", "Failed", "Uploaded", "Uploading"),
+                    customId: S.NullOr(S.String),
                 }))
             });
             return await this.executeAsync(this.requestUploadThing("/api/listFiles", {
